src/ui/public/error_url_overflow/url_overflow_service.js

```diff
@@ -0,0 +1,62 @@
+
+export class UrlOverflowService {
+  constructor() {
+    const key = 'error/url-overflow/url';
+    const store = window.sessionStorage || {
+      getItem() {},
+      setItem() {},
+      removeItem() {},
+    };
+
+    // FIXME: Couldn't find a way to test for browser compatibility without
+    // complex redirect and cookie based "feature-detection" page, so going
+    // with user-agent detection for now.
+    this._ieLike = /(;MSIE |Edge\/\d)/.test(window.navigator.userAgent);
+
+    this._val = store.getItem(key);
+    this._sync = () => {
+      if (this._val == null) store.removeItem(key);
+      else store.setItem(key, this._val);
+    };
+  }
+
+  failLength() {
+    return this._ieLike ? 2000 : 25000;
```

Comment:
The idea was that all browsers have a limit, unfortunately they aren't documented. [This page](https://boutell.com/newfaq/misc/urllength.html) says that firefox and safari both support more than 60,000 characters, but I figured kibana would start to fail long before we got there, so I chose 25,000 as our "max". We could always bump or remove this number.